frame/staking/src/pallet/mod.rs

```diff
@@ -49,6 +49,7 @@ use crate::{
 };
 
 const STAKING_ID: LockIdentifier = *b"staking ";
+pub(crate) const SPECULATIVE_NUM_SPANS: u32 = 32;
```

Comment:
May be some comment about why we speculate slashing spans would be 32? Is this the maximum it can be?

Reply:
Can it be different for different runtimes?

Reply:
I believe there is no max for the number of spans. I pulled the numbers of current slashing spans and calculated a few metrics over it for Kusama an Polkadot, sharing here if you are interested:

```
Polkadot:
 sum: 8577
 avg: 1.314410480349345
 median: 1
 q75: 1
 q90: 2
 q999: 7.629000000000019

boundaries: [1, 9]
---

Kusama:
 sum: 602
 avg: 3.4626564392410173
 median: 2
 q75: 4
 q90: 8
 q999: 46.43199999999797
 boundaries: [1, 65]

qX is the quantile X, i.e. the value that stands at the position x% of the sorted array with all slashing spans.  
```

32 is a conservative number in Polkadot as you can see in the numbers above (max num spans so far is 9). In Kusama, I believe there were some issues in the past that triggered slashings, and that's why the max spans are larger. Given that we don't want to set the default dispatchable weight too high (to decrease the changes of a transaction being left out of a block due to weight over-provisioning that would be refunded post dispatch) and that the chances of both 1) a call require a unbond+withdraw and 2) slashing spans are > 32 , I believe this is a sensitive number. Even if the number of slashing spans is larger than the spaculative number and the withdraw is required before unbonding, there is no critical issues with it.

Reply:
I will add a comment explaining what `SPECULATIVE_NUM_SPANS` are and why we set it to 32.